src/System.Diagnostics.Process/src/System/Diagnostics/ProcessManager.Linux.cs

```diff
@@ -110,21 +110,17 @@ internal static ProcessModuleCollection GetModules(int processId)
         /// </summary>
         internal static ProcessInfo CreateProcessInfo(int pid, ReusableTextReader reusableReader = null)
         {
-            if (reusableReader == null)
-            {
-                reusableReader = new ReusableTextReader();
-            }
+            reusableReader ??= new ReusableTextReader();
+            bool readStatFile = Interop.procfs.TryReadStatFile(pid, out Interop.procfs.ParsedStat stat, reusableReader);
+            bool readStatusFile = Interop.procfs.TryReadStatusFile(pid, out Interop.procfs.ParsedStatus status, reusableReader);
 
-            Interop.procfs.ParsedStat stat;
-            return Interop.procfs.TryReadStatFile(pid, out stat, reusableReader) ?
-                CreateProcessInfo(ref stat, reusableReader) :
-                null;
+            return readStatFile || readStatusFile ? CreateProcessInfo(ref stat, ref status, reusableReader) : null;
```

Comment:
> readStatFile || readStatusFile

I don't understand this.  How do we create the process info if we don't have a valid `stat`?  We can do so without a valid `status`, but `stat` appears necessary, no?

Reply:
`Pid` is an input parameter, we already have that, so that's how we access `/proc/[pid]/stat`. Now what if reading `/stat` fails and `/status` doesn't ? You want to fail it as well ?

Reply:
That's not how it's populated:
https://github.com/dotnet/corefx/blob/d3911035f2ba3eb5c44310342cc1d654e42aa316/src/System.Diagnostics.Process/src/System/Diagnostics/ProcessManager.Linux.cs#L129

Reply:
>Now what if reading /stat fails and /status doesn't ? You want to fail it as well ?

Yup

Reply:
> That's not how it's populated:
> https://github.com/dotnet/corefx/blob/d3911035f2ba3eb5c44310342cc1d654e42aa316/src/System.Diagnostics.Process/src/System/Diagnostics/ProcessManager.Linux.cs#L129
> 
> 
> Note that `pid` currently isn't passed into CreateProcessInfo.

Ah. Ok when you pass it forward, you read from the object itself.

Reply:
Done.